src/bvh/bvh.rs

```diff
@@ -299,13 +366,13 @@ impl BoundingHierarchy for BVH {
     /// ```
     fn traverse<'a, T: Bounded>(&'a self, ray: &Ray, shapes: &'a [T]) -> Vec<&T> {
         let mut indices = Vec::new();
-        self.root.traverse_recursive(ray, &mut indices);
+        self.nodes[0].traverse_recursive(&self.nodes, ray, &mut indices);
         let mut hit_shapes = Vec::new();
         for index in &indices {
             let shape = &shapes[*index];
-            if ray.intersects_aabb(&shape.aabb()) {
-                hit_shapes.push(shape);
-            }
+            // if ray.intersects_aabb(&shape.aabb()) {
+            hit_shapes.push(shape);
```

Comment:
I haven't been sure for some time about it.

I just removed the `if` completely now. The reason is that there is only one shape per leaf now, and if you get to the leaf during traversal, the shape's AABB has already been tested against.